packages/plugin-electron-deliver-minidumps/minidump-queue.js

```diff
@@ -16,6 +18,19 @@ module.exports = class MinidumpQueue {
     return this._minidumps[0]
   }
 
+  push (minidump) {
+    // if _minidumps has not yet been lazy-loaded, this minidump will be picked
+    // up automatically when the list loads
+    if (this._minidumps && !this._seen.has(minidump.minidumpPath)) {
+      this._seen.add(minidump.minidumpPath)
+      this._minidumps.push(minidump)
+    }
+  }
+
+  hasSeen (minidumpPath) {
+    return this._seen.has(minidumpPath)
+  }
```

Comment:
I don't really understand why this method would be necessary, can you explain what it will be used for?

As I understand it, any new minidumps can be pushed to the queue and then `peek` will return them, so I'm not sure why it would be necessary to check if a minidump has been seen already

Reply:
That's fair - this check is used for determining whether a minidump found after a background crash event is actually new or is already being uploaded. Having it exist means that we can skip unnecessarily writing an extra event info file, which has to be done before pushing it into the queue.